staging/src/k8s.io/client-go/plugin/pkg/client/auth/openstack/openstack.go

```diff
@@ -145,11 +156,33 @@ func newOpenstackAuthProvider(clusterAddress string, config map[string]string, p
 		}
 	}
 
-	// TODO: read/persist client configuration(OS_XXX env vars) in config
+	configByte, err := json.Marshal(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal config: %s", err)
+	}
+
+	var authOpt gophercloud.AuthOptions
+	err = json.Unmarshal(configByte, &authOpt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config to AuthOptions: %s", err)
+	}
+
+	getter := tokenGetter{}
+	// not empty
+	if (authOpt != gophercloud.AuthOptions{}) {
+		// TODO: remove this when gophercloud supports marshal this field
+		endpoint, ok := config["identityEndpoint"]
```

Comment:
this is another sign that this probably isn't the right approach.

if you want to populate options from inline config, I'd suggest doing it explicitly, like this:

```
gophercloud.AuthOptions{
	IdentityEndpoint: config["identityEndpoint"],
	Username: config["username"],
	Password: config["password"],
	DomainName: config["name"],
	TenantID: config["tenantId"],
	TenantName: config["tenantName"],
}
```

by accepting "identityEndpoint", you are already diverging from a standard auth options json blob, so being explicit doesn't seem bad.

Reply:
> you are already diverging from a standard auth options json blob, so being explicit doesn't seem bad.

Make sense.